models/image.py

```diff
@@ -640,20 +656,29 @@ def save(self, filename=None):
                     if not array_name.endswith('.fits'):
                         array_name += '.fits'
                     extensions.append(array_name)
+                    if not single_file:
+                        files_written[array_name] = extpath
 
-            if single_file and not filename.endswith('.fits'):
-                filename += '.fits'
+            if single_file:
+                files_written = files_written['.image.fits']
+                if not self.filepath.endswith('.fits'):
+                    self.filepath += '.fits'
 
         elif format == 'hdf5':
             # TODO: consider writing a more generic utility to save_image_file that handles either fits or hdf5, etc.
             raise RuntimeError("HDF5 format is not yet supported.")
         else:
             raise ValueError(f"Unknown image format: {format}. Use 'fits' or 'hdf5'.")
 
-        self.filepath = filename
```

Comment:
I sort of like the idea that the filepath is only filled when the file is saved. If you can work around that assumption I think it will be good for us (for example, we can do an event hook that doesn't let you save to DB before filepath is filled and that it points to an existing file).